File: NX8.5/NX8.5_32_8.5.3.3/UGOPEN/SampleNXOpenApplications/.NET/InteropNXOpenWithUFWrap/InteropNXOpenWithUFWrap.cs
/*=============================================================================

                    Copyright (c) 2004 UGS PLM Solutions
                    Unpublished - All rights reserved

===============================================================================
File description :
This sample program demostrats the interoprability between NX Open and NX Open
UF API. 

 
=============================================================================
*/
using System;
using System.IO;
using NXOpen;
using NXOpen.Utilities;
using NXOpen.UF;
//********************************************************************************
//
//Steps
//
//1. Create a new part file			using		NXOpen
//2. Query the part name 			using		NXOpen.UF
//3. Create Line					using		NXOpen
//4. Query line data				using		NXOpen.UF
//5. Create Arc						using		NXOpen.UF
//6. Query  Arc parameters			using		NXOpen
//7. Save the Part					using		NXOpen.UF
//*********************************************************************************
// How to get NX Open object from NX Open .Net Tag
//		NXOpen.Arc NxArc= (NXOpen.Arc)NXOpen.Utilities.NXObjectManager.Get(arc);
//
// How to get NX Open .Net Tag from NXOpen Object
//			NXOpen.Line line1  
//			NXOpen.Tag line_tag= line1.Tag
//*********************************************************************************

namespace NXOpenTestCase
{
    class InteropNXOpenWithUFWrap
    {
        static void Main(string[] args)
        {
            //Get the NX Open API
            Session theNXAutomationSession = Session.GetSession();
            // Get the NX Open API
            //NXOpen.UI theUI = UI.GetUI();
            //Get the UF Session using NX Open UF API
            UFSession theNXOpenNetSession = UFSession.GetUFSession();
            try
            {
                //Declartions
                string name;

                // Intilize the NX Open UF Session, This will not needed once UF 
				// will get intilized in Session. 
                //theNXOpenNetSession.UF.Initialize();

                //Create new part using NX Open UF API
                //string part_name = "CreateLine2";
                //int units =2; 
                //theNXOpenNetSession.Part.New(part_name, units, out part);

                //Create New Part file using NX Open API
                NXOpen.Part myPart= theNXAutomationSession.Parts.NewDisplay("InteropNXOpenWithUFWrap",NXOpen.Part.Units.Millimeters);
                NXOpen.Tag part=myPart.Tag;
                //Query the part name using NX Open API
                theNXOpenNetSession.Part.AskPartName(part, out name);
                
                // Create Line using NX Open APIs
                NXOpen.Point3d point3d1 = new Point3d(-2.45197396411307, 3.58206319143819, 0);
                NXOpen.Point3d point3d2 = new Point3d(5.32514590979158, -1.0012853802839, 0);
                NXOpen.Line line1 = theNXAutomationSession.Parts.Work.Curves.CreateLine(point3d1, point3d2);

                //Ask line data using NX Open UF API
                NXOpen.UF.UFCurve.Line line_coords;
                theNXOpenNetSession.Curve.AskLineData(line1.Tag,out line_coords);
			
                // Create Arc using NXOpen UF API
                NXOpen.Tag arc, wcs;
                NXOpen.UF.UFCurve.Arc arc_coords = new NXOpen.UF.UFCurve.Arc();		
		
                // Fill out the data structure 
                arc_coords.start_angle = 0.0;
                arc_coords.end_angle = 3.0;
                arc_coords.arc_center=new double[3];
                arc_coords.arc_center[0] = 0.0;
                arc_coords.arc_center[1] = 0.0;
                arc_coords.arc_center[2] = 1.0;
                arc_coords.radius = 2.0;
		
                // Create a ARC using NXOpen UF API
                theNXOpenNetSession.Csys.AskWcs(out wcs);
                theNXOpenNetSession.Csys.AskMatrixOfObject(wcs,out arc_coords.matrix_tag);
                theNXOpenNetSession.Curve.CreateArc(ref arc_coords,out arc);

                // Create NX Open Arc Object using NXOpen UF arc Tag
                Arc NxArc= (Arc)NXOpen.Utilities.NXObjectManager.Get(arc);

                //Get the Arc parameters using NX Open APIs
                double start_angle= NxArc.StartAngle;
                double end_angle= NxArc.EndAngle;
                NXOpen.Point3d arc_center=NxArc.CenterPoint;

                // Save the Part using NXOpen UF API
                theNXOpenNetSession.Part.Save();
             }
            catch(NXException e)
            {	
                Console.WriteLine("Exception is: {0}", e.Message);
            }
        }
    }
}

File: NX1847_NX1851_NX1855_NX1859_NX1863_NX1867/UGOPEN/SampleNXOpenApplications/Java/InteropNXOpenWithUFWrap/InteropNXOpenWithUFWrap.java
/*==================================================================================================

            Copyright (c) 2004 UGS PLM Solutions.
                   Unpublished - All rights reserved

====================================================================================================
File description:
Example to show how to create Arc.
 
*/

import nxopen.*;
import nxopen.uf.*;
import java.io.PrintWriter;
import java.io.StringWriter;


public class InteropNXOpenWithUFWrapper 
{
   
    public static void main(String[] args) throws Exception
    {
        Session theSession=null ;
        UFSession theUFSession=null ;

        try
        {
            theSession = (Session)SessionFactory.get("Session");
            theUFSession = (UFSession)SessionFactory.get("UFSession");
            UFCurve ufCurve = theUFSession.curve();
            UFCurve.Arc ufArc = new UFCurve.Arc();
            UFPart ufPart = theUFSession.part();
            UFCsys ufCsys = theUFSession.csys();
            Part part1 = theSession.parts().newDisplay("InteropNXOpenWithUFWrapper", Part.Units.MILLIMETERS);

            Point3d arcStartPoint = new Point3d(-2.45197396411307, 3.58206319143819, 0.0);
            Point3d arcEndPoint = new Point3d(5.32514590979158, -1.0012853802839, 0.0);
            Line line1 = theSession.parts().work().curves().createLine(point3d1, point3d2);
            UFCurve.Line line_coords = theUFSession.curve().askLineData(line1.tag());
            /* Fill out the data structure */
            ufArc.startAngle = 0.0;
            ufArc.endAngle = 3.0;
            ufArc.arcCenter=new double[3];
            ufArc.arcCenter[0] = 0.0;
            ufArc.arcCenter[1] = 0.0;
            ufArc.arcCenter[2] = 1.0;
            ufArc.radius = 2.0;

            Tag wcsData = ufCsys.askWcs();
            ufArc.matrixTag = ufCsys.askMatrixOfObject(wcsData);
            Tag arcTag = ufCurve.createArc(ufArc);
            Arc NXArc = (Arc) theSession.taggedObjectManager().get(arcTag);

            double startAngle = NXArc.startAngle();
            double endAngle = NXArc.endAngle();
            Point3d arcCenter = NXArc.centerPoint();

        }
        catch (Exception ex)
        {
            if(theUFSession!=null)
            {
                StringWriter s = new StringWriter();
                PrintWriter p = new PrintWriter(s);
                theUFSession.ui().openListingWindow();
                p.println("Caught exception " + ex );
                ex.printStackTrace(p);
                theUFSession.ui().writeListingWindow("\nFailed");
                //theUFSession.ui().writeListingWindow("\n"+ex.getMessage());
                theUFSession.ui().writeListingWindow("\n"+s.getBuffer().toString());
            }
        
        
        }
    }

     public static int getUnloadOption() { 
        return BaseSession.LibraryUnloadOption.IMMEDIATELY; 
    }
}
